### PR TITLE
New Entry: General WSDL

### PR DESCRIPTION
**Pull request by SSwiniarski**
### Description

From Target Keyword List

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/wsdl/wsdl.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+- `<service>`: Defines the collection of ports supported by the web service.
+- `<documentation>`: Human-readable comments that can be included within any other WSDL element.
+
+## Structure
+
+The structure of a WSDL document looks like this:
+
+```pseudo
+<definitions>
+   <types>
```
Comment (Dusch4593) [inline content/general/concepts/wsdl/wsdl.md:40]: Is the indentation supposed to be 2 or 3 spaces (currently this is 3 spaces).

**Review by Dusch4593 — CHANGES_REQUESTED** [review]
@SSwiniarski Just one comment/suggestion from me for this one 😄

**Comment by SSwiniarski** [comment]
@Dusch4593: RE: your comment, I changed it to two spaces.

**Review by Dusch4593 — APPROVED** [review]
@SSwiniarski LGTM! Let's get one more review for this one 😄

File: content/general/concepts/wsdl/wsdl.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  - 'Development'
+  - 'Tags'
+CatalogContent:
+  - 'learn-html'
+  - 'paths/front-end-engineer-career-path'
+---
+
+**WSDL (Web Services Description Language)** is a specification for an [XML](https://www.codecademy.com/resources/docs/general/xml) document that describes a web service. A WSDL document is a machine-readable description of how a service can be called, what parameters it expects, and what type of data structures are returned by it. [WSDL Version 2.0](https://www.w3.org/TR/wsdl/) was released by the [World Wide Web Consortium](https://www.w3.org/standards/) (W3C) in 2007, but support for this version of the specification is still poorly implemented, with much web development software only supporting [WSDL Version 1.1](https://www.w3.org/TR/2001/NOTE-wsdl-20010315).
+
+WSDL documents are ususally constructed and read by web development software, and developers don't often need to interact with these files directly.
```
Comment (caupolicandiaz) [inline content/general/concepts/wsdl/wsdl.md:18]:
```suggestion
WSDL documents are usually constructed and read by web development software, and developers don't often need to interact with these files directly.
```
```suggestion
WSDL documents are ususally constructed and read by web development software, and developers don't often need to interact with these files directly.
```

Comment (caupolicandiaz) [inline content/general/concepts/wsdl/wsdl.md:18]:
```suggestion
WSDL documents are usually constructed and read by web development software, and developers don't often need to interact with these files directly.
```
```suggestion
WSDL documents are ususally constructed and read by web development software, and developers don't often need to interact with these files directly.
```

File: content/general/concepts/wsdl/wsdl.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+## Elements
+
+A WSDL document is made up of the following elements:
+
+- `<definitions>`: The root element defining the name of the service, declares namespaces used in the document, and contains all the other elements.
+- `<types>`: The data types used in the form of XML schemas.
+- `<message>`: Description of the data exchanged between the web service and its consumers.
+- `<portType>`: Combines multiple messages to form a complete transaction.
```
Comment (caupolicandiaz) [inline content/general/concepts/wsdl/wsdl.md:27]: I'm not familiar with this topic but the other sources I reviewed seem to have a more expansive definition in which the portType outlines the operations that can be performed and the relevant messages required. Am I misinterpreting the text here?

Comment (caupolicandiaz) [inline content/general/concepts/wsdl/wsdl.md:27]: I'm not familiar with this topic but the other sources I reviewed seem to have a more expansive definition in which the portType outlines the operations that can be performed and the relevant messages required. Am I misinterpreting the text here?

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@SSwiniarski just a few small comments

File: content/general/concepts/wsdl/wsdl.md (hunk 3 above)
Comment (SSwiniarski) [reply content/general/concepts/wsdl/wsdl.md:27]: I was aiming to give a general overview, not to get too deep in the weeds on syntax, especially as these files are generally machine-generated. Do you think I need to go into more detail?

Comment (caupolicandiaz) [reply content/general/concepts/wsdl/wsdl.md:26]: That's fair, let's leave it as is

**Review by caupolicandiaz — APPROVED** [review]
@SSwiniarski all set

**Comment by Dusch4593** [comment]
Live entry here: https://www.codecademy.com/resources/docs/general/wsdl

**Pull request merged**